packages/material-ui/src/ExpansionPanel/ExpansionPanel.js

```diff
@@ -82,11 +81,9 @@ class ExpansionPanel extends React.Component {
   constructor(props) {
     super();
     this.isControlled = props.expanded != null;
-    this.state = {};
-    if (!this.isControlled) {
-      // not controlled, use internal state
-      this.state.expanded = props.defaultExpanded !== undefined ? props.defaultExpanded : false;
```

Comment:
We never check if the state is undefined. `isControlled` is the point of truth. This was just a verbose boolean cast.